reader_test.go

```diff
@@ -78,3 +79,56 @@ func TestUnilogReader(t *testing.T) {
 		}
 	}
 }
+
+type OurReader struct {
+	input chan []byte
+}
+
+func (or *OurReader) Read(p []byte) (n int, err error) {
+	bs, ok := <-or.input
+	if !ok {
+		return 0, io.EOF
+	}
+	return copy(p, bs), nil
+}
+
+func TestReaderAlone(t *testing.T) {
+	t.Run("basic", func(t *testing.T) {
```

Comment:
I think you could `t.Parallel()` these (or make them separate test functions, really - they don't share any setup code...)